packages/ra-core/src/form/useAugmentedForm.ts

```diff
@@ -81,6 +87,18 @@ export const useAugmentedForm = <RecordType = any>(
     // notify on invalid form
     useNotifyIsFormInvalid(form.control, !disableInvalidFormNotification);
 
+    const recordFromLocation = useRecordFromLocation();
+    const recordFromLocationApplied = useRef(false);
+    const { reset } = form;
+    useEffect(() => {
+        if (recordFromLocation && !recordFromLocationApplied.current) {
+            reset(merge({}, record, recordFromLocation), {
```

Comment:
You should probably merge `recordFromLocation` with `defaultValuesIncludingRecord` instead of `record`, otherwise the `defaultValues` are discarded.